### DISCUSSION TITLE
BundlesAction, strange instruction zipped=true

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-equinox/p2#105] [Q&A]
I'm currently plying around with the meta-data generation of Tycho and noticed that for a BundleInfo created from a directory, P2 generates a strange

```
<instructions size='2'>
        <instruction key='zipped'>
          true
        </instruction>
```

This is done here https://github.com/eclipse-equinox/p2/blob/master/bundles/org.eclipse.equinox.p2.publisher.eclipse/src/org/eclipse/equinox/p2/publisher/eclipse/BundlesAction.java in Line 326.

I don't understand what this touchpoint is supposed to do? Should it "zip the directory" when published?

**Comment by mickaelistria** [comment]
> Should it "zip the directory" when published?

If I read the code correctly, this is added when `Eclipse-BundleShape: dir` in the MANIFEST.MF, so I believe a better interpretation would be that the jar must be unzipped and turned into a directory during installation.
